### PR TITLE
internal: don't shut up the compiler when it says the code's buggy

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

File: crates/rust-analyzer/src/dispatch.rs [hunk 1]
```diff
@@ -10,14 +10,30 @@ use crate::{
     LspError, Result,
 };
 
+/// A visitor for routing a raw JSON request to an appropriate handler function.
+///
+/// Most requests are read-only and async and are handled on the threadpool
+/// (`on` method).
+///
+/// Some read-only requests are latency sensitive, and are immediately handled
+/// on the main loop thread (`on_sync`). These are typically typing-related
```
Comment (matklad) [inline crates/rust-analyzer/src/dispatch.rs:19]: @Veykril ``[`RequestDispatcher::on_sync`]`` doesn't work for me. Is it our bug, or is this just isn't supported by rustdoc?

Comment (bjorn3) [reply crates/rust-analyzer/src/dispatch.rs:19]: Should work I think.

Comment (Veykril) [reply crates/rust-analyzer/src/dispatch.rs:19]: Ye I'm pretty sure that should work with rustdoc. Odd that we don't resolve this 🤔

Comment (Veykril) [reply crates/rust-analyzer/src/dispatch.rs:19]: Oh right this is https://github.com/rust-analyzer/rust-analyzer/issues/9694

**Pull request merged**